store/sqlstore/post_store.go

```diff
@@ -399,26 +399,26 @@ func (s *SqlPostStore) GetEtag(channelId string, allowFromCache bool) store.Stor
 	})
 }
 
-func (s *SqlPostStore) Delete(postId string, time int64, deleteByID string) store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
+func (s *SqlPostStore) Delete(postId string, time int64, deleteByID string) *model.AppError {
 
-		appErr := func(errMsg string) *model.AppError {
-			return model.NewAppError("SqlPostStore.Delete", "store.sql_post.delete.app_error", nil, "id="+postId+", err="+errMsg, http.StatusInternalServerError)
-		}
+	appErr := func(errMsg string) *model.AppError {
+		return model.NewAppError("SqlPostStore.Delete", "store.sql_post.delete.app_error", nil, "id="+postId+", err="+errMsg, http.StatusInternalServerError)
+	}
 
-		var post model.Post
-		err := s.GetReplica().SelectOne(&post, "SELECT * FROM Posts WHERE Id = :Id AND DeleteAt = 0", map[string]interface{}{"Id": postId})
-		if err != nil {
-			result.Err = appErr(err.Error())
-		}
+	var post model.Post
+	err := s.GetReplica().SelectOne(&post, "SELECT * FROM Posts WHERE Id = :Id AND DeleteAt = 0", map[string]interface{}{"Id": postId})
+	if err != nil {
+		return appErr(err.Error())
+	}
 
```

Comment:
I might be misreading this, but before the change the code didn't exit early in case of an error (line 412) and continued to the line below. In the new code - it aborts with a return. Was this change intentional?

Reply:
There is a subtle change in the behavior here, but I think is ok. Before this change it was checking if the post exists and is not deleted, but anyway try to update the post. Now only try to update the post if the post exists and is not deleted.

Reply:
Oki doki. In that case - LGTM!